docs/kb/intro-functions.md

```diff
@@ -8,25 +8,324 @@ cta:
+
+To subscribe to a specific vCenter Server event, we can apply a filtering to our Knative Trigger like the example below:
+
+```
+apiVersion: eventing.knative.dev/v1
+kind: Trigger
+metadata:
+  name: veba-ps-echo-trigger
+  namespace: vmware-functions
+spec:
+  broker: default
+  filter:
+    attributes:
+       type: com.vmware.event.router/event
+       subject: VmPoweredOffEvent
+  subscriber:
+    ref:
+      apiVersion: serving.knative.dev/v1
+      kind: Service
+      name: kn-ps-echo
+```
+
+`veba-ps-echo-trigger`: The name of the Knative trigger
+
+The value of this field:
+
+- must not conflict with an existing trigger
+- should not contain special characters, e.g. "$" or "/"
+- should represent the intent of the trigger, e.g. "tag" or "tagging"
+
+`default`: The name of Knative broker. For VEBA with Embedded Knative Broker, the value will be `default`
+
+`VmPoweredOffEvent`: The name of the vCenter Server event Id. Please refer to [vCenter Events](vcenter-events) for list of supported events.
+
+`kn-ps-echo`: The name of the Knative Service
+
+> **Note:** Today, a single Knative Trigger can only filter on one vCenter Server event. To associate multiple vCenter Server events to a given Knative Service, you simply create a Knative Trigger for each event.
+
+## Knative Combined Service and Trigger
+
+To simplify Knative function deployment, we can also combine the multiple manifest files into a single file. In the example below, the `function.yaml` contains both the Knative Service and Trigger definition.
+
+```
+apiVersion: serving.knative.dev/v1
+kind: Service
+metadata:
+ name: kn-ps-echo
+ namespace: vmware-functions
+spec:
+ template:
+  spec:
+   containers:
+    - image: projects.registry.vmware.com/veba/kn-ps-echo:latest
+---
+apiVersion: eventing.knative.dev/v1
+kind: Trigger
+metadata:
+  name: veba-ps-echo-trigger
+  namespace: vmware-functions
+spec:
+  broker: default
+  filter:
+    attributes:
+       type: com.vmware.event.router/event
+       subject: VmPoweredOffEvent
+  subscriber:
+    ref:
+      apiVersion: serving.knative.dev/v1
+      kind: Service
+      name: kn-ps-echo
+      namespace: vmware-functions
+```
+
+To deploy the Knative Service/Trigger, just run:
+
+```
+kubectl apply -f function.yaml
+```
+
+To undeploy the Knative Service/Trigger, just run:
+
+```
+kubectl delete -f function.yaml
+```
+
+## Knative Secrets
+
+Knative functions also support secrets or sensitive information which are passed in as part of the function deployment. Within the function, you can then access the secrets using an environmental variable and the name that the secret reference was created with.
+
+A secrets file (without file extension) should be created that contains the sensitive values including the structure you wish to process within your function. For example, you can encode your secrets into a JSON structure which means you can use the language specific JSON parser to easily extract out specific values.
```

Comment:
Since you use env KEY name when creating the secret, the file extension actually does not matter.